internal/integration/integration_test.go

```diff
@@ -53,316 +69,323 @@ func TestIntegration(t *testing.T) {
 		OnlyLocalProvenance: false,
 	}
 
-	keys := util.GenerateExposureKeys(3, -1, false)
+	for _, tc := range cases {
```

Comment:
For a table driven test, `t.Run` might be better, as it would make debug much easier

Reply:
Also drop a t.Parallel() in the t.Run() so the subtests run in parallel too

Reply:
done

Reply:
Why is that a problem?

Reply:
Added it in the latest commit. Somehow I had misinterpreted the error I was seeing